src/plugins/discover/public/application/angular/helpers/row_formatter.tsx

```diff
@@ -6,24 +6,28 @@
  * Side Public License, v 1.
  */
 
-import { template } from 'lodash';
+import React, { Fragment } from 'react';
+import ReactDOM from 'react-dom/server';
 import { MAX_DOC_FIELDS_DISPLAYED } from '../../../../common';
 import { getServices, IndexPattern } from '../../../kibana_services';
 
-function noWhiteSpace(html: string) {
-  const TAGS_WITH_WS = />\s+</g;
-  return html.replace(TAGS_WITH_WS, '><');
+interface Props {
+  defPairs: Array<[string, unknown]>;
 }
-
-const templateHtml = `
-  <dl class="source truncate-by-height">
-    <% defPairs.forEach(function (def) { %>
-      <dt><%- def[0] %>:</dt>
-      <dd><%= def[1] %></dd>
-      <%= ' ' %>
-    <% }); %>
-  </dl>`;
-export const doTemplate = template(noWhiteSpace(templateHtml));
+const TemplateComponent = ({ defPairs }: Props) => {
+  return (
+    <dl className={'source truncate-by-height'}>
+      {defPairs.map((pair, idx) => (
+        <Fragment key={idx}>
+          <dt>{pair[0]}:</dt>
+          <dd
+            dangerouslySetInnerHTML={{ __html: `${pair[1]}` }} // eslint-disable-line react/no-danger
```

Comment:
"dangerous" rendering is equivalent to the existing `<dd><%= def[1] %></dd>` syntax used by the lodash template engine.

Reply:
Could we please nevertheless add a comment in the code why we consider `dangerouslySetInnerHTML` here safe, i.e. in this case I assume because it's always running via field formatters? (cc @kertal ). I try to make sure we have an inline comment on any place we use dangerouslySetInnerHTML so we can later easily refer to why we thought it's safe :)

Reply:
Great idea, will do!

Reply:
Done in [`0a32a38` (#98072)](https://github.com/elastic/kibana/pull/98072/commits/0a32a38ee24e656ee21e9f5d61beb4e059886c84)